composites/Plugin/SnippetEditor/components/SnippetEditor.js

```diff
@@ -352,17 +352,24 @@ class SnippetEditor extends React.Component {
 		// Strip multiple spaces and spaces at the beginning and end.
 		description = stripSpaces( description );
 
-		const mappedData = {
+		const shortenedBaseUrl = baseUrl.replace( "https://", "" );
+
+		let mappedData = {
 			title: this.processReplacementVariables( originalData.title ),
-			url: baseUrl.replace( "https://", "" ) + originalData.slug,
+			url: shortenedBaseUrl + originalData.slug,
 			description: description,
 		};
 
+		const context = {
+			shortenedBaseUrl,
+		};
+
 		// The mapping by the passed mapping function should happen before measuring.
 		if ( mapDataToPreview ) {
-			return mapDataToPreview( mappedData, originalData );
+			return mapDataToPreview( mappedData, context );
```

Comment:
I find confusing that this `mapDataToPreview` prop (which passes a function) has the same name of the `mapDataToPreview` function in the class itself. Also, it gets passed two arguments but the passed mapping function is `mapEditorDataToPreview` and it accepts one argument, unless I'm missing something.

Reply:
You're completely right, this one is my mistake, I forgot to push changes on the worpdpress-seo sibling pr 😓

Reply:
I've discussed the name of the function with Irene. Unfortunately, both names describe exactly what they do, so renaming it quickly makes the new name suboptimal. 

If the reviewer has a better suggestion for the name, you're welcome to suggest them, but we couldn't think of one.

Reply:
I'd suggest to rename the prop that passes the function to start with, currently it's:
`mapDataToPreview={ mapEditorDataToPreview }`
and rename it to
`mapEditorDataToPreview={ mapEditorDataToPreview }`

This would require a small change also in the plugin. Alternatively, not destructure the props and use `this.props.mapDataToPreview` could make the code more readable.